crates/cashu/src/nuts/nut11/mod.rs

```diff
@@ -894,14 +918,15 @@ impl SwapRequest {
 
         // Add all input secrets in order
         for proof in self.inputs() {
-            let secret = proof.secret.to_string();
-            msg_to_sign.push_str(&secret);
+            msg_to_sign.push_str(&proof.secret.to_string());
+            msg_to_sign.push_str(&proof.c.to_hex());
         }
 
-        // Add all output blinded messages in order
+        // Add all blank outputs in order if they exist
         for output in self.outputs() {
-            let message = output.blinded_secret.to_string();
-            msg_to_sign.push_str(&message);
+            msg_to_sign.push_str(&output.amount.to_string());
+            msg_to_sign.push_str(&output.keyset_id.to_string());
```

Comment:
The current iteration of [the spec](https://github.com/cashubtc/nuts/pull/302) removed the keyset_id.